Test/Callbacks/test_callbacks.py

```diff
@@ -0,0 +1,29 @@
+import netket as nk
+
+SEED = 3141592
+
+
+def _run_vmc(**kwargs):
+    nk.random.seed(SEED)
+    g = nk.graph.Hypercube(length=8, n_dim=1)
+    hi = nk.hilbert.Spin(s=0.5, graph=g)
+
+    ma = nk.machine.RbmSpin(hilbert=hi, alpha=1)
+    ma.init_random_parameters(sigma=0.01, seed=SEED)
+
+    ha = nk.operator.Ising(hi, h=1.0)
+    sa = nk.sampler.MetropolisLocal(machine=ma)
+
+    op = nk.optimizer.Sgd(ma, learning_rate=0.1)
+
+    vmc = nk.Vmc(hamiltonian=ha, sampler=sa, optimizer=op, n_samples=500)
+    es = nk.callbacks.EarlyStopping(**kwargs)
+    vmc.run(300, callback=es)
+
+
+def test_earlystopping_with_patience():
+    _run_vmc(patience=10)  # Runs for ~50 iterations only
```

Comment:
This is not testing that your code works as expected.
It's just testing that it does not errors.

Doesn't this work?
```python
vmc = setup_vmc(...)
es = nk.callbacks.EarlyStopping(patience=10)
es._best_value = 1e-10
vmc.run(20, callback=es)
assert vmc.step_value == 10
```

Reply:
Ok, I had to expose the `best_value` attribute, but now I test that it works as intended.